src/core/utils/url.js

```diff
@@ -23,11 +28,60 @@ export function removeUndefinedProps(object: Object): Object {
 export function addQueryParams(
   urlString: string,
   queryParams: { [key: string]: ?string | number } = {},
+  _config: typeof config = config,
 ): string {
+  let adjustedQueryParams = { ...queryParams };
+
+  if (
+    _config.get('enableFeatureUseUtmParams') &&
+    typeof queryParams.src !== 'undefined'
+  ) {
+    adjustedQueryParams = {
+      ...queryParams,
+      // Use UTM parameters instead of `src`, according to the PRD.
+      utm_source: DEFAULT_UTM_SOURCE,
+      utm_medium: DEFAULT_UTM_MEDIUM,
+      utm_content: queryParams.src,
+      src: undefined,
+    };
+  }
+
   const urlObj = url.parse(urlString, true);
   // Clear search, since query object will only be used if search property
   // doesn't exist.
   urlObj.search = undefined;
-  urlObj.query = removeUndefinedProps({ ...urlObj.query, ...queryParams });
+  urlObj.query = removeUndefinedProps({
+    ...urlObj.query,
+    ...adjustedQueryParams,
+  });
+
   return url.format(urlObj);
 }
+
+export function getQueryParametersForAttribution(
+  location: ReactRouterLocationType,
+  _config: typeof config = config,
+): Object {
+  let params = { src: location.query.src };
+
+  if (_config.get('enableFeatureUseUtmParams')) {
+    if (typeof location.query.src !== 'undefined') {
+      params = {
+        // Use UTM parameters instead of `src`. This allows us to easily update
+        // all query parameters for attribution.
+        utm_source: DEFAULT_UTM_SOURCE,
+        utm_medium: DEFAULT_UTM_MEDIUM,
+        utm_content: location.query.src,
+      };
+    } else {
+      params = {
+        utm_campaign: location.query.utm_campaign,
```

Comment:
how do we know these properties (`utm_campaign`, `utm_content`, etc) are set on `location.query`?  Are they added somewhere else?

Reply:
`location.query` reflects the actual query string in the URL bar. This is when users set their own UTM tracking.